src/Bicep.Core.IntegrationTests/ResourceTests.cs

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System.Linq;
+using Bicep.Core.Diagnostics;
+using Bicep.Core.Semantics;
+using Bicep.Core.UnitTests.Assertions;
+using Bicep.Core.UnitTests.Utils;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Bicep.Core.IntegrationTests
+{
+    [TestClass]
+    public class ResourceTests
+    {
+        [TestMethod]
+        public void NestedResources_child_cycle_is_detected_correctly()
+        {
+            var program = @"
+resource parent 'My.RP/parentType@2020-01-01' = {
+  name: 'parent'
+  properties: {
+    style: child.properties.style
```

Comment:
I'm surprised that this works. Any weird scoping issues pop up with this?

Reply:
well this is a negative test. The compiler raises an error about a circular dependency - so it doesn't really work 😆

Reply:
ah ok great - I just saw the code sample and didn't look where it was coming from - whoops!